applications/GeoMechanicsApplication/tests/cpp_tests/custom_conditions/test_U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -21,95 +21,355 @@
+                                       Element::GeometryType::Pointer    pNeighbourGeometry,
+                                       ModelPart&                        rModelPart)
+{
     // set properties of the condition
     Vector absorbing_factors = ZeroVector(2);
     absorbing_factors[0]     = 1.0;
     absorbing_factors[1]     = 1.0;
+    pCondition->SetValue(ABSORBING_FACTORS, absorbing_factors);
+    pCondition->SetValue(VIRTUAL_THICKNESS, 100.0);
+    // add neighbour element to condition
 
-    cond->SetValue(ABSORBING_FACTORS, absorbing_factors);
-    cond->SetValue(VIRTUAL_THICKNESS, 100.0);
-
-    // create mock neighbour element
     auto p_neighbour_prop = rModelPart.CreateNewProperties(1);
-    p_neighbour_prop->SetValue(POROSITY, 1.0);
+    p_neighbour_prop->SetValue(POROSITY, 0.0);
     p_neighbour_prop->SetValue(DENSITY_WATER, 1000);
     p_neighbour_prop->SetValue(DENSITY_SOLID, 2000);
 
-    auto neighbour_node_1 = rModelPart.CreateNewNode(3, 1.0, 1.0, 0.0);
-    auto neighbour_node_2 = rModelPart.CreateNewNode(4, 0.0, 1.0, 0.0);
+    const auto p_neighbour_element =
+        Kratos::make_intrusive<MockElement>(1, pNeighbourGeometry, p_neighbour_prop);
+    rModelPart.AddElement(p_neighbour_element);
 
-    Element::GeometryType::Pointer p_geometry =
-        Kratos::make_shared<Kratos::Quadrilateral2D4<Element::NodeType>>(
-            node_1, node_2, neighbour_node_1, neighbour_node_2);
-    const auto mock_element = Kratos::make_intrusive<MockElement>(1, p_geometry, p_neighbour_prop);
-    rModelPart.AddElement(mock_element);
-
-    // add neighbour element to condition
     GlobalPointersVector<Element> vector_of_neighbours;
-    vector_of_neighbours.push_back(Kratos::GlobalPointer<Kratos::Element>(mock_element.get()));
-    cond->SetValue(NEIGHBOUR_ELEMENTS, vector_of_neighbours);
+    vector_of_neighbours.push_back(Kratos::GlobalPointer<Kratos::Element>(p_neighbour_element.get()));
+    pCondition->SetValue(NEIGHBOUR_ELEMENTS, vector_of_neighbours);
+
+    p_neighbour_element->SetValue(POROSITY, 0.0);
+    p_neighbour_element->SetValue(DENSITY_WATER, 1000);
+    p_neighbour_element->SetValue(DENSITY_SOLID, 2000);
 
     // Initialize the element
     const auto& r_process_info = rModelPart.GetProcessInfo();
-    cond->Initialize(r_process_info);
+    pCondition->Initialize(r_process_info);
+}
+
+/// <summary>
+/// Sets up a 2D 2N UPwLysmerAbsorbingCondition with a 2D 4N neighbour element
+/// </summary>
+static ModelPart::ConditionType::Pointer SetUpUPwLysmerAbsorbingCondition2D2NCondition(ModelPart& rModelPart)
+{
+    rModelPart.GetProcessInfo().SetValue(DOMAIN_SIZE, 2);
+    rModelPart.AddNodalSolutionStepVariable(DISPLACEMENT);
+    rModelPart.AddNodalSolutionStepVariable(WATER_PRESSURE);
+
+    // Set the element properties
+    auto cond_prop = rModelPart.CreateNewProperties(0);
+
+    auto node_1 = rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
+    auto node_2 = rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
```

Comment:
The names node_x are not used afterwards.
```suggestion
    rModelPart.CreateNewNode(1, 0.0, 0.0, 0.0);
    rModelPart.CreateNewNode(2, 1.0, 0.0, 0.0);
```